docs/reference/analysis/analyzers/lang-analyzer.asciidoc

```diff
@@ -36,1481 +39,515 @@ following types are supported:
 
-The `stem_exclusion` parameter allows you to specify an array
-of lowercase words that should not be stemmed.  Internally, this
-functionality is implemented by adding the
-<<analysis-keyword-marker-tokenfilter,`keyword_marker` token filter>>
-with the `keywords` set to the value of the `stem_exclusion` parameter.
+`there, is, no, wai, that, writer, can, be, tame, and, render, civil`
+
+[source,js]
+--------------------------------------------------------------------------------
+include::../../snippets/analysis/my-english-analyzer.json[]
+--------------------------------------------------------------------------------
+// SENSE: analysis/my-english-analyzer.json
 
-The following analyzers support setting custom `stem_exclusion` list:
+[[configuring-stemming]]
+==== Configuring Stemming
+
+Several language analyzers support disabling stemming for particular words.
+Stemming is the process of reducing words to the base or root form.
+You set the `stem_exclusion` parameter to specify which words should not be
+stemmed. This parameter takes an array of lowercase words.
+For more information, see <<analysis-keyword-marker-tokenfilter,Keyword Marker Token Filter>>.
+
+The following analyzers support the `stem_exclusion` parameter:
 `arabic`, `armenian`, `basque`, `catalan`, `bulgarian`, `catalan`,
 `czech`, `finnish`, `dutch`, `english`, `finnish`, `french`, `galician`,
 `german`, `irish`, `hindi`, `hungarian`, `indonesian`, `italian`, `latvian`, `norwegian`,
 `portuguese`, `romanian`, `russian`, `sorani`, `spanish`, `swedish`, `turkish`.
 
-==== Reimplementing language analyzers
+For example, the following `english` analyzer configuration prevents `civilized` and `way` from
+being stemmed. 
 
-The built-in language analyzers can be reimplemented as `custom` analyzers
-(as described below) in order to customize their behaviour.
+[source,js]
+--------------------------------------------------------------------------------
+include::../../snippets/analysis/configure-english-analyzer-keywords.json[]
+--------------------------------------------------------------------------------
+// SENSE: analysis/configure-english-analyzer-keywords.json
 
-NOTE: If you do not intend to exclude words from being stemmed (the
-equivalent of the `stem_exclusion` parameter above), then you should remove
-the `keyword_marker` token filter from the custom analyzer configuration.
+Using this analyzer to process the string  "There is no way that writers can be 
+tamed and rendered civilized" generates the following tokens:
+
+`there, is, no, way, that, writer, can, be, tame, and, render, civilized`
+
+[source,js]
+--------------------------------------------------------------------------------
+include::../../snippets/analysis/my-english-analyzer.json[]
+--------------------------------------------------------------------------------
+// SENSE: analysis/my-english-analyzer.json
+
+==== Customizing Language Analyzers
```

Comment:
You'd have "Language analyzers" in the ToC, followed by  "Customizing language analyzers". I think that's OK, no?  Otherwise the Language Analyzers page is a bit overwhelming.